src/components/Admin/Admin.js

```diff
@@ -34,109 +35,145 @@ class Admin extends React.Component {
     super(props);
 
     this.state = {
-      index: 0
+      index: 0,
+      showModal: false
     };
 
+    this.closeModal = this.closeModal.bind(this);
+    this.openModal = this.openModal.bind(this);
     this.handleTabChanged = this.handleTabChanged.bind(this);
+    this.restartPipelineAndCloseModal = this.restartPipelineAndCloseModal.bind(this);
```

Comment:
FYI: you can avoid these bind calls by using arrow functions at the class level which will ensure that `this` will bind correctly.

Example:

```js
restartPipelineAndCloseModal = () => {
  this.props.flux.actions.ADMIN.restart_pipeline();
  this.closeModal();
}
```

[More information](https://medium.freecodecamp.org/react-binding-patterns-5-approaches-for-handling-this-92c651b5af56)